### DISCUSSION TITLE
[RFC] New Variant of Progress: type `inline-circle`

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#38119] [RFCs - archive]
![image](https://user-images.githubusercontent.com/27722486/196705076-2534747b-8720-452d-a1da-5875cb86fd0e.png)

## Summary
Progress 组件新增 `inline-circle` 变体

## Motivation
在中后台产品中，我们经常会遇到需要和文字配合，或嵌入其他组件内容内来表示任务进展状态及进度的场景，此时常规的进度条因为尺寸较大难以满足我们的需求。故需要一种内容场景下的进度条变体，以解决此类问题。

## API

`type="inline-circle"`
| 属性        | 说明     | 类型             | 默认值 |
| --- | --- | --- | --- |
| strokeColor | 圆形进度条线的色彩，传入 object 时为渐变         | string \| object | -      |
| text |  进度条右侧文本  | `ReactNode` | - |

剩余属性与 Progress 相同

## Basic Example
![image](https://user-images.githubusercontent.com/27722486/196707692-5ee98623-9814-42e1-8d35-ff8dbf97c5be.png)

```tsx
import React, { useRef } from React;
import { Progress } from 'antd';

export default () => (
  <Progress type="inline-circle" text="代码发布" format={(percent) => `进行中，已完成${percent}%`} />
);
```

### 尺寸
circle 的默认尺寸其实只有 14px，所以在画布中需要留 1px 的边距。线宽为 1.5px，暂不支持修改尺寸。
![image](https://user-images.githubusercontent.com/27722486/196708106-9259482a-a74e-4915-bd86-390a3c98a798.png)

### 进度信息
在上述例子中，进度信息在 Tooltip 中展示，对应 prop 为 Progress 公用的 `format`。hover 热区默认宽度为 28px，圆角为 4px

### 状态
![image](https://user-images.githubusercontent.com/27722486/196710336-b4d7ff02-5254-4380-aa22-058d49c93ef9.png)
可由 `status` 属性改变。

## Drawbacks
No obvious drawbacks yet.

**Comment by afc163** [comment]
`<Progress type="circle" inline />` 如何

会不会还有 `<Progress type="line" inline />` ？

**Reply by MadCcc** [reply]
考虑到这个才叫 `inline-circle` 的～因为设计和 `circle` 很不一样， API 也会有区别，所以在 `type` 这层就区分开了

**Reply by afc163** [reply]
API 有区别的话，就建议新子组件了，不然 TS 定义也不好写。

**Reply by MadCcc** [reply]
Progress 的每个 type 都有独特的 API，感觉可以考虑一起拆一下

**Comment by li-jia-nan** [comment]
mark~

**Comment by afc163** [comment]
这个设计是否要默认实现我是存疑的，1 是这个 hover 背景比较占位置，不一定合适。2 是这种感觉变化比较多，比如要展现一个 Popover 显示更多内容，内建 Tooltip + format 灵活性是不够的。

**Comment by afc163** [comment]
我建议只实现 <img width="44" alt="图片" src="https://user-images.githubusercontent.com/507615/197780799-9827e654-7762-47be-be31-f1c4aa9c0c1e.png"> 这个元素。

**Comment by li-jia-nan** [comment]
我看这个还在讨论中，要么我先不做了，等最终方案确定下来再做
